### DISCUSSION TITLE
What would it take for helioviewer.org to use the WCS information?

### DISCUSSION THREAD

**Discussion by wafels** [Helioviewer-Project/helioviewer.org#506] [Ideas]
Many of the JPEG2000 images are actually manipulated versions of the original data that change the original WCS information so that the image is easier to plot in the helioviewer.org interface. What would it take for helioviewer.org to understand the original FITS WCS information so that we did not need to create images with changed metadata (WCS mainly) so that the interface could plot it? This would simplify the image processing (FITS to JPEG2000) and presumably make overplotting significantly more flexible.

**Comment by dgarciabriseno** [comment]
What kind of changes are made to the original WCS for Helioviewer?

**Reply by wafels** [reply]
Changing the WCS so that CRVAL1,2 is zero. For EIT, changing the pixel size so that all the data looks like it is from 1 AU.

If we kept the original WCS and projected it correctly we wouldn't need to do any of that.

**Reply by dgarciabriseno** [reply]
We could try a less processed image and see how it shows up.

The image already gets shifted for the refpixel and takes CRVAL into account: https://github.com/Helioviewer-Project/api/blob/f93a73ef4246f0d986c11f2c5bf88592d887f633/src/Image/JPEG2000/JP2ImageXMLBox.php#L267

But I can see potential issues extracting tiles. For example if we need the bottom right quarter, extract it from the jpeg2000 by supplying the rectangle top left (50%, 50%) and width/height (50%, 50%). I think right now for everything the refpixel is the center of the image and the coordinate is (0, 0). So the 50/50 works. So if it was shifted we'd have to account for that.

**Comment by wafels** [comment]
See also https://github.com/Helioviewer-Project/helioviewer.org/issues/509
